src/app-layer-parser.c

```diff
@@ -919,6 +918,23 @@ void AppLayerParserTransactionsCleanup(Flow *f, const uint8_t pkt_dir)
     SCLogDebug("start min %"PRIu64, min);
     bool skipped = false;
     // const bool support_files = AppLayerParserSupportsFiles(f->proto, f->alproto);
+    uint32_t tofree_nb = 0;
+    if (total_txs > UINT32_MAX) {
+        SCFree(txl->tx_id_free_list);
+        txl->tx_id_free_list = NULL;
```

Comment:
it doesn't seem to be necessary to free it here?

Reply:
Ok for not freeing.

But we still need to handle this case where we have way too many live transactions (4 billion transactions).
Adding a new local bool to do that.